native/qrusty/src/lib.rs

```diff
@@ -38,6 +40,18 @@ fn image_base64_nif(data: &str, opts: Options) -> Result<(Atom, String), Error>
     Ok((ok(), base64::encode(&bytes)))
 }
 
+#[rustler::nif]
+fn svg_alphanumeric(data: &str, opts: Options) -> Result<(Atom, String), Error> {
+  let qr_bits = encode_bits(data, opts.error_correction.t()).unwrap();
+
+  let code = QrCode::with_bits(qr_bits, opts.error_correction.t()).unwrap();
```

Comment:
I'm not familiar enough with our use of this lib to say that we are actually going to need to do this, especially since the other functions are doing it. But `.unwrap()` is normally reserved for times when you know you are not going to have a failure as it is going to end up raising if you do have the failure case. The safer alternative would be to use `?` instead. 

```suggestion
  let qr_bits = encode_bits(data, opts.error_correction.t())?;

  let code = QrCode::with_bits(qr_bits, opts.error_correction.t())?;
```

Instead of raising when you have the error case, it will just return the error instead.

Now, I know [zigler](https://hex.pm/packages/zigler) will automatically catch exceptions and turn them into an exception on the Elixir side. I don't remember if Rust is doing the same thing or not.

Again, this isn't something that you actually need to do because the other functions in this lib use `.unwrap()`. I just thought it would be worth mentioning in case you anyone here ends up doing more rust and comes across it.

Reply:
Good to know! I do like that better, although we basically never get errors out of this library.